### PR TITLE
fix completion not working with functions again and fix overflow bug …

### PR DESCRIPTION
**Pull request by triclops200**
…introduced by large numbers of completion options

Fixes issue #726

### CODE REVIEW

File: input.lisp [hunk 1]
```diff
@@ -379,12 +384,12 @@ match with an element of the completions."
...
-                                       (< (length str) (length input-line))
-                                       (string/= input-line
-                                                 (subseq str 0 (length input-line)))))
-                                 *input-completions*))
+         (completions (take *maximum-completions* (remove-if (lambda (str)
+                                                               (or (string= str "")
+                                                                   (< (length str) (length input-line))
+                                                                   (string/= input-line
+                                                                             (subseq str 0 (length input-line)))))
+                                                             *input-completions*)))
```
Comment (triclops200) [inline input.lisp]: There should also probably be a deduplication step by wrapping the whole thing in (remove-duplicates).

File: input.lisp [hunk 2]
```diff
@@ -420,58 +425,58 @@ match with an element of the completions."
...
+                               (nth i prompt-lines)
+                               :translate #'translate-id
+                               :size 16)
+         else
+         do (draw-image-glyphs win gcontext font
+                               *message-window-padding*
+                               (prompt-text-y i font *message-window-y-padding*)
+                               (nth (- i prompt-lines-length) completions)
+                               :translate #'translate-id
+                               :size 16))
```
Comment (triclops200) [inline input.lisp]: Auto-formatting from my emacs. Sorry about this. The next block of "changes" is the same.

File: input.lisp [hunk 3]
```diff
@@ -317,7 +318,11 @@ match with an element of the completions."
 (defun read-one-line (screen prompt &key completions (initial-input "") require-match password)
   "Read a line of input through stumpwm and return it. Returns nil if the user aborted."
   (let ((*input-last-command* nil)
-        (*input-completions* completions)
+        (*input-completions* (if (or (functionp completions)
+                                     (and (symbolp completions)
+                                          (fboundp completions)))
+                                 (funcall completions "")
```
Comment (agspathis) [inline input.lisp]: Maybe use `initial-input` here?

Comment (triclops200) [reply input.lisp]: newest commit uses initial-input, good catch.

**Pull request merged**